ext/include/opentelemetry/ext/http/client/http_client_factory.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "opentelemetry/ext/http/client/http_client.h"
+#ifdef WITH_CURL
+#  include "opentelemetry/ext/http/client/curl/http_client_curl.h"
+#endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace http
+{
+namespace client
+{
+class HttpClientFactory
+{
+public:
+#ifdef WITH_CURL
+  static std::shared_ptr<SessionManager> Create()
+  {
+    return std::make_shared<ext::http::client::curl::SessionManager>();
```

Comment:
What are the cases when we need a SessionManager, can it become a private detail of curl client, and be auto-refreshed as-needed? First time you trigger the request/response, you maintain a session. Maybe I'm missing some fundamental reason _why_ the session manager is unique to curl, and why is it required at all? Looking at `prior art`, we were not needing it elsewhere 😄 
https://github.com/microsoft/cpp_client_telemetry/blob/master/lib/http/HttpClient_Curl.cpp
https://github.com/microsoft/cpp_client_telemetry/blob/master/lib/http/HttpClient_Curl.hpp

Reply:
Nice to see reference of `cpp_client_telemetry` now that it is public  :)
So, session Manager is not unique to `curl` but an abstract class which should be implemented for all client libraries. `opentelmetry::ext::http::client::SessionManager` is similar to `MAT_NS_BEGIN::IHttpClient`, and it manages multiple ongoing HTTP requests.
`opentelementry::ext::http::client::Session` logically represent a (possibly) ongoing http request.

Reply:
Please see my response above.